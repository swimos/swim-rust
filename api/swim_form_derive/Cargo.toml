[package]
name = "swim_form_derive"
version = "0.1.0"
authors = ["Swim.AI developers info@swim.ai"]
edition = "2021"

[lib]
name = "swim_form_derive"
proc-macro = true

[dependencies]
either = "1.5.3"
swim_utilities = { path = "../../swim_utilities", features = ["errors"] }
macro_utilities = { path = "../../macro_utilities" }
proc-macro2 = "1.0"
syn = { version = "1.0", features = ["full"] }
quote = "1.0.3"

[dev-dependencies]