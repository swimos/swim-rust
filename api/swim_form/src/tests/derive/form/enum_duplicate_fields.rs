use swim_form_derive::*;

fn main() {
    #[allow(non_snake_case)]
    #[derive(Form)]
    #[form_root(::swim_form)]
    enum Duplicates {
        First {
            #[form(convention = "camel")]
            first_field: i32,
            firstField: i32,
        },
        #[form(fields_convention = "camel")]
        Second {
            first_field: i32,
            #[form(name = "firstField")]
            second_field: i32,
        }
    }
}