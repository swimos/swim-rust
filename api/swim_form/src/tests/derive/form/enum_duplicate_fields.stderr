error: Form field names must be unique. Duplicated names: [firstField]
  --> src/tests/derive/form/enum_duplicate_fields.rs:8:9
   |
8  | /         First {
9  | |             #[form(convention = "camel")]
10 | |             first_field: i32,
11 | |             firstField: i32,
12 | |         },
   | |_________^

error: Form field names must be unique. Duplicated names: [firstField]
  --> src/tests/derive/form/enum_duplicate_fields.rs:13:9
   |
13 | /         #[form(fields_convention = "camel")]
14 | |         Second {
15 | |             first_field: i32,
16 | |             #[form(name = "firstField")]
17 | |             second_field: i32,
18 | |         }
   | |_________^
